ludwig/schema/features/augmentation/utils.py

```diff
@@ -0,0 +1,156 @@
+from dataclasses import field
+from typing import List, Union
+
+from marshmallow import fields, ValidationError
+
+from ludwig.api_annotations import DeveloperAPI
+from ludwig.constants import TYPE
+from ludwig.schema import utils as schema_utils
+from ludwig.utils.registry import Registry
+
+_augmentation_config_registry = Registry()
+
+
+@DeveloperAPI
+def get_augmentation_config_registry() -> Registry:
+    return _augmentation_config_registry
+
+
+@DeveloperAPI
+def register_augmentation_config(name: str, features: Union[str, List[str]]):
+    if isinstance(features, str):
+        features = [features]
+
+    def wrap(cls):
+        for feature in features:
+            augmentation_registry = get_augmentation_config_registry().get(feature, {})
+            augmentation_registry[name] = cls
+            get_augmentation_config_registry()[feature] = augmentation_registry
+        return cls
+
+    return wrap
+
+
+@DeveloperAPI
+def get_augmentation_cls(feature: str, name: str):
+    return get_augmentation_config_registry()[feature][name]
+
+
+@DeveloperAPI
+def get_augmentation_classes(feature: str):
+    return get_augmentation_config_registry()[feature]
+
+
+@DeveloperAPI
+def AugmentationContainerDataclassField(feature_type: str, default=[], description=""):
+    """Custom dataclass field that when used inside a dataclass will allow the user to specify an augmentation
+    config.
+
+    Args:
+        default: The default augmentation config to use.
+        description: The description of the augmentation config.
+
+    Returns: Initialized dataclass field that converts an untyped dict with params to an augmentation config.
```

Comment:
description nit: this converts a *list* to an augmentation

Reply:
done.